### ISSUE TITLE
Keys with spaces and special chars

### ISSUE THREAD

**Issue by franciscoGPS** [maiha/jq.cr#3]
Hi @maiha! This is an awesome shard, thank you for that.

I want to know if there's a way to read and parse a json with keys containing spaces or other special chars. Came to a public API that returns dates with dashes and spaced words as keys, and I haven't been able to properly map the values. 

```
{
      "Time Series (X)": {
              "2022-12-14": {
                  "1. key1" : "val1", 
              },
              "2022-12-13": {
                  "1. key1" : "val1", 
              },
    } 
}
```
So with this, the two main issues I have are:
1.- Neither the `".Time Series (X)"` nor `".Time Series (X).2022-12-14"`  are allowed to be used because the spaces and dashes. Can the be escaped or encoded somehow?

2.- Since the `".Time Series (X)"` is not an array of objects, how could one specify that the requested key, will return a nested Hash(String, String) object?

```
# This wont work, but I can't find an example handling a similar situation.

  Jq.mapping({
    time_series: { Hash(String, String), ".Time Series (X)"},
  })

```

Thanks in advance. So much great work in this shard.

**Comment by maiha** [comment]
Hi @franciscoGPS 

> 1.- Neither the ".Time Series (X)" ...

First, this is a bug. Thank you for reporting. 👍 
I fixed and released `v1.1.0` that accepts quoted keys correctly. Please try quoted keys like this.
```crystal
    Jq.mapping({
      time_series: { Hash(String, Hash(String, String)), %(."Time Series (X)")},
    })
```

> 2.- Since the ".Time Series (X)" is not an array of objects

Please try to cast `Hash(String, Hash(String, String))`.
I wish you the best of luck in getting what you want. ❤️ 
https://github.com/maiha/jq.cr/blob/v1.1.0/spec/mapping/quoted_keys_spec.cr#L23-L45

**Comment by franciscoGPS** [comment]
Oh this is pretty cool. Thank you so much for taking the time for the fix and the new release. This shard is really awesome and easy to use. 
Thank you thank you thank you! 💯 🥇
